code/datums/effects/acid.dm

```diff
@@ -92,3 +96,20 @@
 	if(acid_goopiness <= 0)
 		return TRUE
 	else return FALSE
+
+/datum/effects/acid/proc/handle_weather()
+	SIGNAL_HANDLER
+
+	var/area/A = get_area(src)
+	if(!A)
+		return
+
+	if(SSweather.is_weather_event && locate(A.master) in SSweather.weather_areas)
+		//smothering_strength is 1-10, we use this to take a proportional amount off the stats
+		duration = duration - (duration * (SSweather.weather_event_instance.fire_smothering_strength / 10))
```

Comment:
you really should make these multipliers, into x * 0.1, it's usually not a concern but for procs that are ran more than once in a blue moon it's good practice